src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,90 @@
+ */
+public class Deadline extends Task {
+    private String taskName;
+    private String deadline;
+    private DateTime dateTime;
+
+    /**
+     * Constructor for creating a task
+     *
+     * @param taskName name of task.
+     */
+    public Deadline(String taskName, DateTime dateTime) {
+        super(taskName);
+        this.taskName = taskName;
+        this.deadline = dateTime.getDateTime();
+        this.dateTime = dateTime;
+    }
+
+    /**
+     * Constructor for creating a task based on whether its done or not
+     * @param taskName  name of task.
+     * @param isDone    whether the task is done or not
+     * @param dateTime the date and time of the deadline stored in a DateTime object
+     */
+    public Deadline(String taskName, boolean isDone, DateTime dateTime) {
+        super(taskName);
+        if (isDone) {
+            super.quietlyCompleteTask();
+        }
+        this.taskName = taskName;
+        this.deadline = dateTime.getDateTime();
+        this.dateTime = dateTime;
+
+    }
+
+    /**
+     * taskValidator does nothing if there are no wrong inputs but throws a WrongInputException
+     * if inputs are invalid
+     * @param input the user's string input
+     * @throws WrongInputException which informs the user of the error and actions to take
+     */
+    public static void taskValidator(String input) throws WrongInputException {
+        String[] segementedString = input.split(" /by ");
+        if (segementedString.length < 2) {
+            throw new WrongInputException("/by command is required",
+                    "Use the /by command using 'deadline <name> /by <time>");
+        } else {
+            String deadline = segementedString[1];
+            try {
+                String taskNameDeadline = segementedString[0].substring(Parser.DEADLINEOFFSET);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new WrongInputException("Task name cannot be blank",
+                        "Enter a non-blank deadline task name");
+            }
+
+            String taskNameDeadline = segementedString[0].substring(Parser.DEADLINEOFFSET);
+            if (taskNameDeadline.trim().isEmpty()) {
+                throw new WrongInputException("Task name cannot be blank",
+                        "Enter a non-blank deadline task name");
+            } else if (deadline.trim().isEmpty()) {
+                throw new WrongInputException("for /by <time>, time cannot be blank",
+                        "Please input valid text that are not space");
+            } else if (!DateTimeParser.isValidDateTime(deadline)) {
+                throw new WrongInputException("Invalid date and time format",
+                        DateTimeParser.getValidDateTimeFormat());
+            }
+        }
+    }
+
+    /**
+     * Converts a Deadline task into a string that can be saved
+     * @return  a string that can be saved
```

Comment:
instead of `* @return  a string that can be saved` should be `* @return a string that can be saved` 
there shouldn't be an extra space in the javadocs